openmls/src/treesync/diff.rs

```diff
@@ -135,18 +136,9 @@ impl<'a> TreeSyncDiff<'a> {
         Ok(())
     }
 
-    /// Adds a new leaf to the tree either by filling a blank leaf or by
-    /// extending the tree to the right to create a new leaf, inserting
-    /// intermediate blanks as necessary. This also adds the leaf_index of the
-    /// new leaf to the `unmerged_leaves` of the parent nodes in its direct
-    /// path.
-    ///
-    /// Returns the LeafIndex of the new leaf.
-    pub(crate) fn add_leaf(
-        &mut self,
-        leaf_node: KeyPackage,
-    ) -> Result<LeafIndex, TreeSyncDiffError> {
-        let node = Node::LeafNode(leaf_node.into());
+    /// Find and return the index of either the left-most blank leaf, or, if
+    /// there are no blank leaves, the leaf count.
+    pub(in crate::treesync) fn free_leaf_index(&self) -> Result<LeafIndex, TreeSyncDiffError> {
         // Find a free leaf and fill it with the new key package.
         let leaf_ids = self.diff.leaves()?;
         let mut leaf_index_option = None;
```

Comment:
I guess this is old code but why is there a `continue` in the loop?

Reply:
I'm confused. What loop?

Reply:
Line 150 (I can't comment there because it's old code). Not too important though

Reply:
That continue is there, because we only want the first free spot that we find and there might be more than one blank leaf.

Reply:
Ok, so you want a `break`? Because this isn't doing what you think it is.
But apperently there's no test actually excercising this.
Please add one and fix this.

Reply:
Urgh. Good point. I think I'd rather fix it separately, though. I filed #653, which includes the fix and a test. Let's get that on main, then I'll merge it in here. It's delays this PR by a little, but it doesn't make the PR even bigger than it already is.